pkg/apis/core/taint.go

```diff
@@ -27,10 +27,8 @@ func (t *Taint) MatchTaint(taintToMatch Taint) bool {
 	return t.Key == taintToMatch.Key && t.Effect == taintToMatch.Effect
 }
 
-// taint.ToString() converts taint struct to string in format key=value:effect or key:effect.
+// taint.ToString() converts taint struct to string in format key=value:effect.
+// The value is optional and may be an empty string.
 func (t *Taint) ToString() string {
```

Comment:
Wondering if we should return an error too, if the taint is not of the expected format. This would be helpful for debugging when someone imports this package internally and uses it just like how you found the issue.

Reply:
> if the taint is not of the expected format

I'm sorry, I'm not sure what you mean by "format." Are you referring to the fields of `t`?

From what I understand, all inputs have been valid, but the function was returning an invalid output (missing the `=` character) for a specific valid input (where the `t.Value` field is an empty string).

Reply:
we probably shouldn't return errors from a `ToString()` method.

Reply:
Agreed, I was thinking in terms on how to add a -ve testcase so suggested it.

Reply:
I think a negative testcase would be good. I can amend the PR.